ibis/expr/geospatial.py

```diff
@@ -0,0 +1,181 @@
+"""GEO Spatial Data Types"""
+import collections
+from shapely import geometry
+import ibis.expr.types as ir
+
+
+class GeoSpatial(object):
+    # TODO: maybe it needs geo spatial system information
+    __slots__ = ()
+
+    def __repr__(self):
+        name = self.name.lower()
+        if not self.nullable:
+            return '{}[non-nullable]'.format(name)
+        return name
+
+
+class Point(GeoSpatial):
+    scalar = ir.PointScalar
+    column = ir.PointColumn
+
+    __slots__ = ()
+
+    def _equal_part(self, other, cache=None):
+        return self.x == other.x and self.y == other.y
+
+
+class Line(GeoSpatial):
+    scalar = ir.LineScalar
+    column = ir.LineColumn
+
+    __slots__ = ()
+
+    def _equal_part(self, other, cache=None):
+        return self.points == other.points
+
+
+class Polygon(GeoSpatial):
+    scalar = ir.PolygonScalar
+    column = ir.PolygonColumn
+
+    __slots__ = ()
+
+    def _equal_part(self, other, cache=None):
+        return self.lines == other.lines
+
+
+class MultiPolygon(GeoSpatial):
+    scalar = ir.PolygonScalar
+    column = ir.PolygonColumn
+
+    __slots__ = ()
+
+    def _equal_part(self, other, cache=None):
+        return self.polygons == other.polygons
+
+
+point = Point()
+line = Line()
+polygon = Polygon()
+multi_polygon = MultiPolygon()
+
+
+class Tokens:
+    POINT = 101
+    LINE = 102
+    POLYGON = 103
+    MULTIPOLYGON = 104
+
+
+_TYPE_RULES = (
+    (
+        'point',
+        'line',
+        'polygon',
+        'multipolygon',
+    ), (
+        Tokens.POINT,
+        Tokens.LINE,
+        Tokens.POLYGON,
+        Tokens.MULTIPOLYGON,
+    )
+)
+
+
+class TypeParser(object):
+    """A type parser for geo spatial types."""
+
+    def type(self):
+        """
+        type : geospatial
+
+        geospatial: point | line | polygon | multipolygon
+
+        """
+        if self._accept(Tokens.POINT):
+            if self._accept(Tokens.LPAREN):
+
+                self._expect(Tokens.PRIMITIVE)
+                x = self.tok.value
+
+                self._expect(Tokens.COMMA)
+
+                self._expect(Tokens.PRIMITIVE)
+                y = self.tok.value
+
+                self._expect(Tokens.RPAREN)
+            else:
+                x = 0
+                y = 0
+            return Point(x, y)
```

Comment:
You need to remove the arguments to the constructors here and elsewhere in the parser.

Reply:
ok! thanks!